core-common/src/main/java/org/zalando/nakadi/partitioning/PartitionStrategy.java

```diff
@@ -13,6 +13,11 @@
     String USER_DEFINED_STRATEGY = "user_defined";
     String RANDOM_STRATEGY = "random";
 
+    default List<String> extractEventKeys(EventType eventType, JSONObject event)
+            throws PartitioningException {
+        throw new PartitioningException("Not supported by partition strategy.");
```

Comment:
We should probably even extract it to a different interface, but this will work for now.